File: PracticaDeEjerciciosParcial/Ejercicio5.py
#Catálogo de música
class Autor:
    def __init__(self, nombre):
        self.nombre = nombre
        self.canciones = []

    def agregar_cancion(self, cancion):
        self.canciones.append(cancion)

class Cancion:
    def __init__(self, titulo, autor):
        self.titulo = titulo
        self.autor = autor
        autor.agregar_cancion(self)

class Disco:
    def __init__(self, titulo):
        self.titulo = titulo
        self.canciones = []

    def agregar_cancion(self, cancion):
        self.canciones.append(cancion)

# Ejemplo de uso
autor = Autor("John Lennon")
cancion1 = Cancion("Imagine", autor)
cancion2 = Cancion("Jealous Guy", autor)

disco = Disco("Best of John Lennon")
disco.agregar_cancion(cancion1)
disco.agregar_cancion(cancion2)

print(f"Disco: {disco.titulo}")
for cancion in disco.canciones:
    print(f"- {cancion.titulo} (Autor: {cancion.autor.nombre})")

File: PracticaParcialTS/PDF2/Ejercicio5.ts
//Catálogo de música
class Autor {
    private canciones: Cancion[] = [];

    constructor(public nombre: string) {}

    agregarCancion(cancion: Cancion): void {
        this.canciones.push(cancion);
    }
}

class Cancion {
    constructor(public titulo: string, public autor: Autor) {
        autor.agregarCancion(this);
    }
}

class Disco {
    private canciones: Cancion[] = [];

    constructor(public titulo: string) {}

    agregarCancion(cancion: Cancion): void {
        this.canciones.push(cancion);
    }

    mostrarCanciones(): void {
        console.log(`Disco: ${this.titulo}`);
        this.canciones.forEach(cancion => {
            console.log(`- ${cancion.titulo} (Autor: ${cancion.autor.nombre})`);
        });
    }
}

// Ejemplo de uso
const autor = new Autor("John Lennon");
const cancion1 = new Cancion("Imagine", autor);
const cancion2 = new Cancion("Jealous Guy", autor);

const disco = new Disco("Best of John Lennon");
disco.agregarCancion(cancion1);
disco.agregarCancion(cancion2);

disco.mostrarCanciones();
